src/frontend/src/expr/mod.rs

```diff
@@ -789,11 +789,16 @@ impl ExprImpl {
         if let ExprImpl::Now(_) = self {
             true
         } else if let ExprImpl::FunctionCall(f) = self {
+            // TODO: `now() + interval '1' month` shall not be accepted as const offset
             match f.func_type() {
                 ExprType::Add | ExprType::Subtract => {
                     let (_, lhs, rhs) = f.clone().decompose_as_binary();
                     lhs.is_now_offset() && rhs.is_const()
                 }
+                ExprType::AddWithTimeZone | ExprType::SubtractWithTimeZone => {
+                    let args = f.inputs();
+                    args[0].is_now_offset() && args[1].is_const()
+                }
```

Comment:
This implementation is too weak. The following optimizer rule actually requires `is_now_offset` to be stricter than `WatermarkAnalyzer`. This was true before the PR but is broken.

https://github.com/risingwavelabs/risingwave/blob/e53c9f74d12d44bb39a61e983208d6468221f8f2/src/frontend/src/optimizer/rule/stream/filter_with_now_to_join_rule.rs#L44-L52

For example, `interval '1' month` is allowed by `is_now_offset` but rejected by `WatermarkAnalyzer`. So the following query (inspired by previous `temporal_filter.slt` failure in ci) would fail the `debug_assert` above:
```
create table t1 (v1 timestamp);
create materialized view mv1 as select v1 from t1 where v1 between now() and now() + interval '1 month';
```

It is not hard to update implementation here to keep that invariant. But I prefer to fix the problem without touching too many different components.